src/components/Email/View.jsx

```diff
@@ -25,15 +25,23 @@ const EmailView = ({
           required={true}
           value={email}
           error={!error ? ["Please enter valid email"] : []}
-          placeholder="Enter your email"
+          placeholder={"Enter your email"}
           onChange={(id, value) => {
             setEmail(value);
             const re = /^(([^<>()\[\]\\.,;:\s@"]+(\.[^<>()\[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
             const error = re.test(String(email).toLowerCase());
             setError(error);
-            if (error) {
-              setFormbuilderInputValue(path, blockid, value);
-            }
+            setFormbuilderInputValue(path, blockid, {
+              label: "Email",
+              validation: {
+                required: true,
+              },
+              placeholder: "Enter your email",
+              customErrormessage: "Please enter a valid email",
+              valid: error,
+              touch: true,
```

Comment:
a boolean value should be called something like "isTouched". Not sure what this one is about though.

Reply:
this is for whether the user attempted to enter the field or not.

Reply:
@iFlameing then maybe "touched"?